build_runner/lib/src/asset/file_based.dart

```diff
@@ -8,11 +8,15 @@ import 'dart:io';
 import 'package:build/build.dart';
 import 'package:glob/glob.dart';
 import 'package:path/path.dart' as path;
+import 'package:pool/pool.dart';
 
 import '../package_graph/package_graph.dart';
 import 'reader.dart';
 import 'writer.dart';
 
+/// Pool for async file writes, we don't want to use too many file handles.
```

Comment:
[nit] "Pool for async file operations"

This also wraps reads

Reply:
done